cli/command/utils.go

```diff
@@ -76,6 +77,67 @@ func PrettyPrint(i any) string {
 
 var ErrPromptTerminated = errdefs.Cancelled(errors.New("prompt terminated"))
 
+type promptOptions struct {
+	disableEcho bool
+}
+
+type PromptOptions func(*promptOptions)
+
+func WithHideUserInput(t bool) PromptOptions {
+	return func(p *promptOptions) {
+		p.disableEcho = t
+	}
+}
+
+// PromptForInput requests input from the user.
+//
+// If the user terminates the CLI with SIGINT or SIGTERM while the prompt is
+// active, the prompt will return an empty string ("") with an ErrPromptTerminated error.
+// When the prompt returns an error, the caller should propagate the error up
+// the stack and close the io.Reader used for the prompt which will prevent the
+// background goroutine from blocking indefinitely.
+func PromptForInput(ctx context.Context, ins *streams.In, outs *streams.Out, message string, opts ...PromptOptions) (string, error) {
```

Comment:
- Looks like `outs` could be bare `io.Writer` (as we don't use any of the features provided by `streams.Out`
- nit: perhaps use singular for the `ins` and `outs` variables (`in` and `out` ?); it's more common.
- ☝️ I recalled we also had a `Streams` interface, which could be an alternative to use as argument, but it looks like that's in the `cli/command` package, so a bit on the fence on that one (see comment further below)
- Wondering though if this function should do a check if `outs` is a terminal, and return an error instead otherwise 🤔 ❓ 

Also wondering; is `message` always required, or could we think of situations where the message itself is already printed, and we're only interested in input provided by a user?

In that case, this could be named `WaitForInput()` and/or have a `WithMessage()` option.


Finally; as this is a new function, and `cli/command` is becoming a bit of a grab-bag of things; wondering if we should put this in a separate package; e.g. `prompt.WaitForInput()` or `userinput.Prompt()`.

Reply:
Even considering; if it's a separate package, we could have a separate `WaitforPassword` (`WaitforPassphrase` ?) or `RequestPassphrase` utility that would handle the `disableEcho`.  In that case, we could get rid of the `PromptOptions` type (at least for now), or keep that one internal until we have new options that could warrant introducing more options.

Reply:
I'm not sure if the `message` parameter could have a default value since this is not a prompt like "confirm x" it's more like an open ended question "input x". 

I've also gotten rid of the variadic parameter and instead opted for a utility function to disable input echo's in the terminal. 

I guess one could also check inside this `PromptForInput` function if it's a terminal or not - but thinking about the `PromptForConfirmation` function we also don't check it there. I guess we would need think about a redesign of these two functions and move them into their own package.